contracts/NonfungiblePositionManager.sol

```diff
@@ -76,25 +66,23 @@ contract NonfungiblePositionManager is INonfungiblePositionManager, BaseLiquidit
         returnData = new bytes[](actions.length);
         for (uint256 i; i < actions.length; i++) {
             if (actions[i] == Actions.INCREASE) {
-                (uint256 tokenId, uint256 liquidity, bytes memory hookData, bool claims) =
-                    abi.decode(params[i], (uint256, uint256, bytes, bool));
-                returnData[i] = abi.encode(increaseLiquidity(tokenId, liquidity, hookData, claims, sender));
+                (uint256 tokenId, uint256 liquidity, bytes memory hookData) =
+                    abi.decode(params[i], (uint256, uint256, bytes));
+                returnData[i] = abi.encode(increaseLiquidity(tokenId, liquidity, hookData, sender));
             } else if (actions[i] == Actions.DECREASE) {
-                (uint256 tokenId, uint256 liquidity, bytes memory hookData, bool claims) =
-                    abi.decode(params[i], (uint256, uint256, bytes, bool));
-                returnData[i] = abi.encode(decreaseLiquidity(tokenId, liquidity, hookData, claims, sender));
+                (uint256 tokenId, uint256 liquidity, bytes memory hookData) =
+                    abi.decode(params[i], (uint256, uint256, bytes));
+                returnData[i] = abi.encode(decreaseLiquidity(tokenId, liquidity, hookData, sender));
             } else if (actions[i] == Actions.MINT) {
                 (LiquidityRange memory range, uint256 liquidity, uint256 deadline, address owner, bytes memory hookData)
                 = abi.decode(params[i], (LiquidityRange, uint256, uint256, address, bytes));
-                (BalanceDelta delta, uint256 tokenId) = mint(range, liquidity, deadline, owner, hookData, sender);
-                returnData[i] = abi.encode(delta, tokenId);
+                returnData[i] = abi.encode(mint(range, liquidity, deadline, owner, hookData));
+            } else if (actions[i] == Actions.CLOSE_CURRENCY) {
+                (Currency currency) = abi.decode(params[i], (Currency));
+                returnData[i] = abi.encode(close(currency, sender));
             } else if (actions[i] == Actions.BURN) {
```

Comment:
Now that v4 has combined `decrease` and `collect` into 1, it feels like we no longer need the multi-step process to close out a position? i.e. if decrease decreases down to 0, shouldnt we just automatically burn the position and not need a burn command? similarly if someone calls increase it could mint a new one?
or is this too complicated?